// cypress-mongodb/dist/commands/collection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createCollection = createCollection;
exports.dropCollection = dropCollection;
var validator_1 = require("../utils/validator");
function createCollection(collection, options) {
    var args = {
        uri: Cypress.env('mongodb').uri,
        database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
        collection: collection,
        options: options,
    };
    (0, validator_1.validate)(args);
    return cy.task('createCollection', args).then(function (result) {
        return result;
    });
}
function dropCollection(collection, options) {
    var args = {
        uri: Cypress.env('mongodb').uri,
        database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
        collection: collection,
        options: options,
    };
    (0, validator_1.validate)(args);
    return cy.task('dropCollection', args).then(function (result) {
        return result;
    });
}
